frontend/src/metabase/query_builder/components/saved-question-picker/utils.js

```diff
@@ -22,3 +22,23 @@ export function buildCollectionTree(collections) {
     children: buildCollectionTree(collection.children),
   }));
 }
+
+export const findCollectionByName = (collections, name) => {
+  if (!collections || collections.length === 0) {
+    return null;
+  }
+
+  const collection = collections.find(c => c.schemaName === name);
+
+  if (collection) {
+    return collection;
+  }
+
+  return findCollectionByName(
+    collections
+      .map(c => c.children)
+      .filter(Boolean)
+      .flat(),
+    name,
+  );
+};
```

Comment:
Can it return an unexpected collection if it happened to have the same name?
If so, can we somehow narrow down the search? Maybe we'll need a backend change?

Reply:
This requires some BE change, @dpsutton and I will take a look separately from this PR